source/common/http/http2/codec_impl.h

```diff
@@ -176,6 +176,18 @@ class ConnectionImpl : public virtual Connection, protected Logger::Loggable<Log
     const Network::Address::InstanceConstSharedPtr& connectionLocalAddress() override {
       return parent_.connection_.localAddress();
     }
+    absl::string_view responseDetails() override { return details_; }
+
+    void setDetails(absl::string_view details) {
+      // Figure out whether we can guarantee that details_ is set at
+      // most once. It is probably a mistake to call setDetails()
+      // twice. For now assert that details_ is empty.
+      ASSERT(details_.empty());
+
+      // make a copy of DETAILS to decouple the lifetime of the two
```

Comment:
s/make/Make/
s/DETAILS/details

But given at the end of the day these can basically be static strings, can we switch over to the ConstSingleton and change, comment lifetime requirements (or lock down access to where this is set to be non-public) and avoid the string creation?

Reply:
The strings are already static strings created via ConstSingleton.  However, in comments above, @yanavlasov remarked that this pattern is risky and recommended that I switch to std::string with the implied copy operation.  (I pointed out that the http1 codec has the same issue.)  Are you suggesting that I revert 39a3fa75e ?